### PR TITLE
Update AzureDevOps.yml

### PR DESCRIPTION
**Pull request by pavelhorak**
Replacing usage of PAT with Entra ID service principal via federated identity

### CODE REVIEW

File: .github/workflows/AzureDevOps.yml [hunk 1]
```diff
@@ -4,20 +4,43 @@ on:
...
+        with:
+          client-id: ${{ secrets.ENTRA_APP_CLIENT_ID }}
+          tenant-id: ${{ secrets.ENTRA_APP_TENANT_ID }}
+          allow-no-subscriptions: true
+      - name: Get Azure DevOps token
+        id: get_ado_token
+        run:
+          # The resource ID for Azure DevOps is always 499b84ac-1321-427f-aa17-267ca6975798
+          # https://learn.microsoft.com/azure/devops/integrate/get-started/authentication/service-principal-managed-identity
+          echo "ado_token=$(az account get-access-token --resource 499b84ac-1321-427f-aa17-267ca6975798 --query "accessToken" --output tsv)" >> $GITHUB_ENV
```
Comment (nohwnd) [inline .github/workflows/AzureDevOps.yml:35]: can the resource id be part of you secrets?

Or can the action below use the azure login to carry out the task?

Comment (pavelhorak) [reply .github/workflows/AzureDevOps.yml:35]: sure, but resource id is a [public information](https://learn.microsoft.com/en-us/azure/devops/organizations/accounts/manage-personal-access-tokens-via-api?view=azure-devops#configure-a-quickstart-application) so there isn't any real need hiding it.

**Review by nohwnd — APPROVED** [review]

File: .github/workflows/AzureDevOps.yml (hunk 1 above)
Comment (nohwnd) [inline .github/workflows/AzureDevOps.yml:35]: nit: 
```suggestion
          echo "ADO_TOKEN=$(az account get-access-token --resource 499b84ac-1321-427f-aa17-267ca6975798 --query "accessToken" --output tsv)" >> $GITHUB_ENV
```

File: .github/workflows/AzureDevOps.yml [hunk 2]
```diff
@@ -4,20 +4,43 @@ on:
...
       - uses: pavelhorak/github-actions-issue-to-work-item@main
         env:
-          ado_token: "${{ secrets.ADO_PERSONAL_ACCESS_TOKEN }}"
+          ado_token: "${{ env.ado_token }}"
```
Comment (nohwnd) [inline .github/workflows/AzureDevOps.yml:38]: nit: 

```suggestion
          ado_token: "${{ env.ADO_TOKEN }}"
```

**Pull request merged**